docs/tracing.md

```diff
@@ -0,0 +1,74 @@
+# Tracing
+
```

Comment:
I'd start this with a brief introduction into what tracing is/what its useful for here. Maybe also settle who the audience is. This seems like a tool that's mainly useful for maintainers, as it requires knowing what files/modules the code sits in/require recompilation.

Reply:
I've added an introduction
```
Firecracker implements a framework for instrumentation based tracing.

Instrumentation based tracing as defined by [Sheng Liang on usenix.org](https://www.usenix.org/legacy/publications/library/proceedings/coots99/full_papers/liang/liang_html/node9.html):

> There are two ways to obtain profiling information: either statistical sampling or code instrumentation. Statistical sampling is less disruptive to program execution, but cannot provide completely accurate information. Code instrumentation, on the other hand, may be more disruptive, but allows the profiler to record all the events it is interested in. Specifically in CPU time profiling, statistical sampling may reveal, for example, the relative percentage of time spent in frequently-called methods, whereas code instrumentation can report the exact number of time each method is invoked.

The focus on tracing in Firecracker is to improve debugability.
```

Reply:
Mh, this covers the "why is it useful" part, but not really the "what does it do" part, e.g. it assumes the reader already knows what "instrumentation based tracing" is. I think we can start with a description of what happens if we enable tracing. Some example log outputs maybe, and after that go into the details of why its useful (not sure we really need the comparison to `perf`, but I am also not opposed to having it)

Reply:
I've added an example at the end of the doc.

Reply:
I still think there should be a brief description of it here at the beginning tbh. Something like "Enabling Firecracker's tracing feature will add additional logging statements to each function entry- and exit-point, printing out the time passed between them. This allows the user to debug problems such as deadlocks or high latencies by quickly identifying which function calls are elongated. For an example, see the end of this document."

Reply:
I've added to the introduction with
```text
Enabling tracing adds logs output at each function entry and exit.
This especially supports debugging problems that relate to deadlocks or
high latencies by quickly identifying elongated function calls.
```